src/genie/libs/parser/asa/show_route.py

```diff
@@ -102,229 +111,129 @@ def cli(self, output=None):
             out = output
 
         ret_dict = {}
-        routes = source_protocol = ''
-        index = 1
-
-        # S* 0.0.0.0 0.0.0.0 via 10.16.251.1, outside
-        # S 0.0.0.1 0.0.0.0 [10/5] via 10.16.255.1, outside
-        p1 = re.compile(
-            r'^\s*(?P<code>(?!is)(?!via)[\w\*]+)\s*(?P<network>\d+.\d+.\d+.\d+)\s*'
-            '(?P<subnet>\d+.\d+.\d+.\d+)\s*(\[(?P<route_preference>[\d\/]+)\])?\s*'
-            '(?P<route_check>[\S]*)\s*(?P<next_hop>\d+.\d+.\d+.\d+),\s*'
-            '(?P<context_name>[\S\s]+)$')
-
-        # via 10.16.251.2, pod1000
-        p2 = re.compile(
-            r'^(?P<network>\d+.\d+.\d+.\d+)?\s*?(?P<route_check>[\S\s]*)\s'
-            '(?P<next_hop>\d+.\d+.\d+.\d+),\s*(?P<context_name>[\S\s]+)$')
-
-        # C 10.10.1.2 255.255.254.0 is directly connected, outside
-        p3 = re.compile(
-            r'^\s*(?P<code>(?!is)(?!via)[\w\*]+)\s*(?P<network>\d+.\d+.\d+.\d+)\s*'
-            '(?P<subnet>\d+.\d+.\d+.\d+)\s*(\[(?P<route_preference>[\d\/]+)\])?\s*'
-            '(?P<route_check>[\S\s]*)\s*,\s*(?P<context_name>[\S\s]+)$')
-
-        # is directly connected, pod2002
-        # connected by VPN (advertised), admin
-        p4 = re.compile(
-            r'^(?P<route_check>[\S\s]*),\s*(?P<context_name>[\S\s]+)$')
-
-        # V 10.10.1.4 255.255.255.255
-        p5 = re.compile(
-            r'^\s*(?P<code>(?!is)(?!via)[\w\*]+)\s*(?P<network>\d+.\d+.\d+.\d+)\s*'
-            '(?P<subnet>\d+.\d+.\d+.\d+)\s*(\[(?P<route_preference>[\d\/]+)\])?\s*'
-            '(?P<context_name>\S+)?$')
-
-        for line in out.splitlines():
-            line = line.strip()
-
-            # S* 0.0.0.0 0.0.0.0 via 10.16.251.1, outside
-            # S 0.0.0.1 0.0.0.0 [10/5] via 10.16.255.1, outside
-            m = p1.match(line)
+        out = re.sub(r'(?ms)(Codes:.+?)replicated\sroute', '', out, re.MULTILINE)
+        r_routes = re.compile(r'(?:^\S{1,4})\s(?:.*)\r?\n(?:^\ +.*\r?\n)*', re.MULTILINE)  # All routes (split)
```

Comment:
could you please add the line of sample output to one line above of the regex patterns, 
same comment for the following patterns,
thanks!

Reply:
Hi, I did a little bit of refactoring on this PR. Hopefully it will cover the issues found.
Cheers
P